pkg/server/secret.go

```diff
@@ -23,159 +16,9 @@ const (
-		if r.Body != nil {
-			defer r.Body.Close()
-		}
-
-		switch r.Method {
-		case http.MethodGet:
-			secrets, err := getSecrets(namespace, kube)
-			if err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				w.Write([]byte(err.Error()))
-				glog.Errorf("Secrets query error: %v", err)
-				return
-			}
-
-			secretsBytes, err := json.Marshal(secrets)
-			if err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				w.Write([]byte(err.Error()))
-				glog.Errorf("Secrets json marshal error: %v", err)
-				return
-			}
-
-			w.Header().Set("Content-Type", "application/json")
-			w.WriteHeader(http.StatusOK)
-			w.Write(secretsBytes)
-		case http.MethodPost:
-			secret, err := parseSecret(namespace, r.Body)
-			if err != nil {
-				w.WriteHeader(http.StatusBadRequest)
-				w.Write([]byte(err.Error()))
-				glog.Errorf("Secret unmarshal error: %v", err)
-				return
-			}
-
-			if err := createSecret(namespace, kube, secret); err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				w.Write([]byte(err.Error()))
-				glog.Errorf("Secret create error: %v", err)
-				return
-			}
-
-			glog.Infof("Secret %s created", secret.GetName())
-			w.WriteHeader(http.StatusAccepted)
-		case http.MethodPut:
-			secret, err := parseSecret(namespace, r.Body)
-			if err != nil {
-				w.WriteHeader(http.StatusBadRequest)
-				w.Write([]byte(err.Error()))
-				glog.Errorf("Secret unmarshal error: %v", err)
-				return
-			}
-
-			if err := updateSecret(namespace, kube, secret); err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				w.Write([]byte(err.Error()))
-				glog.Errorf("Secret update error: %v", err)
-				return
-			}
-
-			glog.Infof("Secret %s updated", secret.GetName())
-			w.WriteHeader(http.StatusAccepted)
-		case http.MethodDelete:
-			secret, err := parseSecret(namespace, r.Body)
-			if err != nil {
-				w.WriteHeader(http.StatusBadRequest)
-				w.Write([]byte(err.Error()))
-				glog.Errorf("Secret unmarshal error: %v", err)
-				return
-			}
-
-			if err := deleteSecret(namespace, kube, secret); err != nil {
-				w.WriteHeader(http.StatusInternalServerError)
-				w.Write([]byte(err.Error()))
-				glog.Errorf("Secret %s delete error: %v", secret.GetName(), err)
-				return
-			}
-
-			glog.Infof("Secret %s deleted", secret.GetName())
-			w.WriteHeader(http.StatusAccepted)
-		default:
-			w.WriteHeader(http.StatusBadRequest)
-			return
-		}
-	}
-}
-
-func getSecrets(namespace string, kube kubernetes.Interface) ([]faastypes.Secret, error) {
-	secrets := []faastypes.Secret{}
-	selector := fmt.Sprintf("%s=%s", secretLabel, secretLabelValue)
-
-	res, err := kube.CoreV1().Secrets(namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: selector})
-
-	if err != nil {
-		return secrets, err
-	}
-
-	for _, item := range res.Items {
-		secret := faastypes.Secret{
-			Name: item.Name,
-		}
-		secrets = append(secrets, secret)
+	handler := handlers.SecretsHandler{
```

Comment:
It isn't clear to me why we make `SecretsHandler` public instead of using the already public factory method `MakeSecretHandler`

Reply:
Yeah Lucas, now I can see that due to your feedback, when I wrote this I didn't thought that actually secrets are generic openfaas feature, and not specific for the operator. Will switch to that :+1: